packages/flutter/test/rendering/viewport_test.dart

```diff
@@ -1144,4 +1144,45 @@ void main() {
       '   its intrinsic dimensions.\n',
     );
   });
+
+  testWidgets('Handles Infinity - Infinity on iOS', (WidgetTester tester) async {
+    // regression test for https://github.com/flutter/flutter/issues/45866
+    final TargetPlatform oldTargetPlatform = debugDefaultTargetPlatformOverride;
+    debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: CustomScrollView(
+          slivers: <Widget>[
+            SliverFillRemaining(
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                mainAxisAlignment: MainAxisAlignment.center,
+                children: <Widget>[
+                  GridView(
+                    shrinkWrap: true,
+                    gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
+                        crossAxisCount: 3,
+                        childAspectRatio: 3,
+                        mainAxisSpacing: 3,
+                        crossAxisSpacing: 3),
+                    children: const <Widget>[
+                      Text('a'),
+                      Text('b'),
+                      Text('c'),
+                    ],
+                  ),
+                ],
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+
+    expect(find.text('b'), findsOneWidget);
```

Comment:
A quick comment on how this is testing the math in the change too.

Reply:
I've added another test that more directly tests the method.  Leaving this one as well since we want it to work.